anghammarad-client-node/test/main.test.ts

```diff
@@ -0,0 +1,66 @@
+import {Anghammarad} from "../src/main";
+import {NotifyParams, RequestedChannel} from "../src/interfaces";
+
+describe("The messageJson function", () => {
+  // Pass something in here so that we don't bother instantiating a client
+  // each time we run the tests
+  const client = new Anghammarad({} as any);
+
+  const defaultParams: NotifyParams = {
+    subject: "subject",
+    message: "message",
+    actions: [{ cta: "cta", url: "url" }],
+    target: { Stack: "stack" },
+    channel: RequestedChannel.All,
+    sourceSystem: "source",
+    topicArn: "arn",
+  };
+
+  it("sets message as provided", () => {
+    expect(client.messageJson(defaultParams)).toContain(`"message":"message"`);
```

Comment:
I'm not a huge fan of checking the JSON string like this, since it depends on the JSON being in a certain format (no spaces) even though that isn't a property we want to enforce here.

Would it be better to parse the response JSON and inspect the resulting value? If we think about the property we're actually checking here, it is that the produced JSON contains a message field with the value `"message"`. Testing that directly rather than checking for a substring may be less fragile while perhaps better conveying the intent to the reader.

There's a few comments on this subject and it's a relatively minor point so no worries either way. Have a think though, see if you agree? Very happy to chat about it.

Reply:
The more I look at this, the more I think these tests aren't doing anything useful. They were added to replicate the scala client but the `messageJson` in this case is far simpler so we're largely just testing that `JSON.stringify` can turn a JS object into JSON... The only change is that the `sourceSystem` key of the params becomes `sender` in the JSON object.

Reply:
I've updated them for now but maybe we can either remove them or consider what is more important to test in a subsequent PR